### ISSUE TITLE
ListObjectsArgs and GetObjectArgs cannot be used in Mocks

### ISSUE THREAD

**Issue by gulimran** [minio/minio-java#1008]
When setting up a mock Minio client:
```
@RunWith(MockitoJUnitRunner.class)

@Mock
private MinioClient minioClient;

ListObjectsArgs listObjectsArgs = ListObjectsArgs.builder().bucket(environment).build();
when(mockMinioClient.listObjects(listObjectsArgs)).thenReturn(result);
```

this mock does not match the argument to Minio client in production code because there is no equals and hashcode implemented for ListObjectsArgs.  Same issue for all other arguments.

**Comment by balamurugana** [comment]
@gulimran Patches are welcome for the requirement.

**Comment by balamurugana** [comment]
Fixed by #1068
